src/apps/climsoft/services/station_service.py

```diff
@@ -0,0 +1,205 @@
+import logging
+from typing import List
+from sqlalchemy.orm.session import Session
+from opencdms.models.climsoft import v4_1_1_core as models
+from src.apps.climsoft.schemas import station_schema
+from fastapi.exceptions import HTTPException
+
+logger = logging.getLogger("ClimsoftStationService")
+logging.basicConfig(level=logging.INFO)
+
+
+class FailedCreatingStation(Exception):
+    pass
+
+
+class FailedGettingStation(Exception):
+    pass
+
+
+class FailedGettingStationList(Exception):
+    pass
+
+
+class FailedUpdatingStation(Exception):
+    pass
+
+
+class FailedDeletingStation(Exception):
+    pass
+
+
+class StationDoesNotExist(Exception):
+    pass
+
+
+def create(db_session: Session, data: station_schema.CreateStation) -> station_schema.Station:
+    try:
+        station = models.Station(**data.dict())
+        db_session.add(station)
+        db_session.commit()
+        return station_schema.Station.from_orm(station)
+    except Exception as e:
+        db_session.rollback()
+        logger.exception(e)
+        raise FailedCreatingStation("Failed creating station.")
+
+
+def get(db_session: Session, station_id: str) -> station_schema.Station:
+    try:
+        station = db_session.query(models.Station).filter_by(stationId=station_id).first()
+
+        if not station:
+            raise HTTPException(status_code=404, detail="Station does not exist.")
+
+        return station_schema.Station.from_orm(station)
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception(e)
+        raise FailedGettingStation("Failed getting station.")
+
+
+def query(
+        db_session: Session,
+        station_id: str = None,
+        station_name: str = None,
+        wmoid: str = None,
+        icaoid: str = None,
+        latitude: float = None,
+        longitude: float = None,
+        qualifier: str = None,
+        elevation: str = None,
+        geolocation_method: str = None,
+        geolocation_accuracy: str = None,
+        opening_datetime: str = None,
+        closing_datetime: str = None,
+        country: str = None,
+        authority: str = None,
+        admin_region: str = None,
+        drainage_basin: str = None,
+        waca_selection: bool = None,
+        cpt_selection: bool = None,
+        station_operational: bool = None,
+        limit: int = 25,
+        offset: int = 0
+) -> List[station_schema.Station]:
+    """
```

Comment:
Please take a look at the docstring to suggest how the query conditions can be improved